pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -344,6 +346,55 @@ public static AggregationFunctionType getAggregationFunctionType(String function
     }
   }
 
+  public static int compareFunctionParametersForStarTree(AggregationFunctionType functionType,
+      Map<String, Object> configuration1, Map<String, Object> configuration2) {
+    switch (functionType) {
+      case DISTINCTCOUNTHLL:
+      case DISTINCTCOUNTRAWHLL: {
+        // Compare the log2m value accounting for defaults
+        if ((MapUtils.isEmpty(configuration1) || !configuration1.containsKey(Constants.HLL_LOG2M_KEY))
+            && (MapUtils.isEmpty(configuration2) || !configuration2.containsKey(Constants.HLL_LOG2M_KEY))) {
+          return 0;
+        }
+        if ((MapUtils.isEmpty(configuration1) || !configuration1.containsKey(Constants.HLL_LOG2M_KEY))) {
+          return Integer.compare(CommonConstants.Helix.DEFAULT_HYPERLOGLOG_LOG2M, Integer.parseInt(String.valueOf(
+              configuration2.get(Constants.HLL_LOG2M_KEY))));
+        }
+        if ((MapUtils.isEmpty(configuration2)) || !configuration2.containsKey(Constants.HLL_LOG2M_KEY)) {
+          return Integer.compare(Integer.parseInt(String.valueOf(configuration1.get(Constants.HLL_LOG2M_KEY))),
+              CommonConstants.Helix.DEFAULT_HYPERLOGLOG_LOG2M);
+        }
+        return Integer.compare(Integer.parseInt(String.valueOf(configuration1.get(Constants.HLL_LOG2M_KEY))),
+            Integer.parseInt(String.valueOf(configuration2.get(Constants.HLL_LOG2M_KEY))));
+      }
+
+      case DISTINCTCOUNTHLLPLUS:
+      case DISTINCTCOUNTRAWHLLPLUS: {
+        // Only p value needs to be compared; HyperLogLogPlus with any sp value can be merged as long as p value is the
+        // same
+        if ((MapUtils.isEmpty(configuration1) || !configuration1.containsKey(Constants.HLLPLUS_ULL_P_KEY))
+            && (MapUtils.isEmpty(configuration2) || !configuration2.containsKey(Constants.HLLPLUS_ULL_P_KEY))) {
+          return 0;
+        }
+        if ((MapUtils.isEmpty(configuration1) || !configuration1.containsKey(Constants.HLLPLUS_ULL_P_KEY))) {
+          return Integer.compare(CommonConstants.Helix.DEFAULT_HYPERLOGLOG_PLUS_P,
+              Integer.parseInt(String.valueOf(configuration2.get(Constants.HLLPLUS_ULL_P_KEY))));
+        }
+        if ((MapUtils.isEmpty(configuration2) || !configuration2.containsKey(Constants.HLLPLUS_ULL_P_KEY))) {
+          return Integer.compare(Integer.parseInt(String.valueOf(configuration1.get(Constants.HLLPLUS_ULL_P_KEY))),
+              CommonConstants.Helix.DEFAULT_HYPERLOGLOG_PLUS_P);
+        }
+        return Integer.compare(Integer.parseInt(String.valueOf(configuration1.get(Constants.HLLPLUS_ULL_P_KEY))),
+            Integer.parseInt(String.valueOf(configuration2.get(Constants.HLLPLUS_ULL_P_KEY))));
+      }
+
+      // By default, assume that the rest of the aggregation functions do not have relevant configurations to be checked
+      // for a star-tree index
+      default:
+        return 0;
```

Comment:
While this is definitely true for some aggregation functions like `COUNT` / `MIN` / `MAX` / `SUM` / `AVG` / `MINMAXRANGE` / `DISTINCTCOUNTBITMAP` / `PERCENTILEEST`, it's a little more nuanced for certain other aggregation functions like `DISTINCTCOUNTULL` and `SUMPRECISION` where if there is a mismatch in the query time aggregation parameter value (different `p` value for `DISTINCTCOUNTULL`, different precision value for `SUMPRECISION`) and the star-tree index aggregation parameter value. In these cases, there will never be a query error since intermediate results with any parameter value can be merged but the result could have lower / higher precision than desired by the user. On second thought, I'm wondering if it might just be better to do a stricter parameter match for these aggregation functions too.

Reply:
@yashmayya @Jackie-Jiang I am late to the conversation but I wanted to draw your attention to a use case which might not be immediately apparent.  There are some aggregates that are in the order of megabytes when serialised to binary, for example, the Tuple sketch.  The Theta sketch is typically 512kb to 1mb at K=16.
Users might want to store a high precision StarTree index and lower the precision at query time - especially when many values are to be merged.  This results in smaller sketches being returned to the broker from the intermediate servers and yields better query performance.  The user is also able to have higher precision for smaller subsets of the data and pay these costs if necessary.   Using a single StarTree for both cases prevents excessive storage overheads.

I realise that this is a probably a "nuanced" use case as stated above - but in the context of these changes, I don't see a way to avoid storage overheads and duplicate StarTrees at different precisions.  My understanding is that the `canUseStarTree` function avoids any changes in precision to the end user and always tries to honour the precision passed in the query aggregation function.  We have been treating the query parameter as a "best effort" value and not as a guarantee.

Reply:
To add to the above - "best effort" only makes sense when going from higher to lower precision and not the other way around.  What is your opinion on using this logic for certain sketch types like CPC?
For example, for the CPC sketch, `canUseStarTree` changes from `_lgNominalEntries == Integer.parseInt(String.valueOf(lgK))` to `_lgNominalEntries >= Integer.parseInt(String.valueOf(lgK))`?

Reply:
Thanks for weighing in here @davecromberge! I think your explanation makes perfect sense, and the suggested logic for matching a CPC sketch query aggregation to a star-tree index looks good. I'd be happy to make that change if you'd like, or to do a review if you'd prefer to author the change yourself.

Reply:
Thank you @yashmayya!  I am busy making the same changes for other Apache Datasketches sketches and would appreciate your review when I create the PR.